python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -744,6 +750,29 @@ def do_evaluate(
                 flags=plc.strings.regex_flags.RegexFlags.DEFAULT,
             )
             return Column(plc.strings.contains.contains_re(column.obj, prog))
+        elif self.name == pl_expr.StringFunction.Slice:
+            child, expr_offset, expr_length = self.children
+            column = child.evaluate(df, context=context, mapping=mapping)
+            if isinstance(expr_offset, Literal) and isinstance(expr_length, Literal):
```

Comment:
You can avoid the indentation (and check with `python -O`) because you've raised in the constructor, and do:

```
assert isinstance(expr_offset, Literal) and isinstance(expr_length, Literal)
```